osu.Game/Overlays/BeatmapSet/BeatmapBadgePill.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Extensions.LocalisationExtensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Localisation;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+
+#nullable enable
+
+namespace osu.Game.Overlays.BeatmapSet
+{
+    public abstract class BeatmapBadgePill : CompositeDrawable
+    {
+        [Resolved]
+        protected OsuColour Colours { get; private set; } = null!;
+
+        [Resolved(canBeNull: true)]
+        protected OverlayColourProvider? ColourProvider { get; private set; }
```

Comment:
Exposing as protected seems unnecessary.

Reply:
Was just consistent with the one above it.

Reply:
but nothing is using it?

hopefully these go away naturally once `BadgeText` and `BadgeColour` become plain properties anyway.

Reply:
Sure.